### PR TITLE
8657 - added unit tests for generateSupportLink

### LINKED ISSUE #8657: Support link should go to correct docs of what version is installed
**Setup**
- Rancher version: v2.7.2
- Browser type & version: Firefox SLE 150200

**Describe the bug**
Get support forwards to v2.7.1 page.

**To Reproduce**
Rancher Home -> Commercial Support -> Find out more about SUSE Rancher Support

**Result**
Opens  https://www.suse.com/suse-rancher/support-matrix/all-supported-versions/rancher-v2-7-1/

**Expected Result**
Should show info related to v2.7.2. 

**Screenshots**

**Additional context**

### PR DESCRIPTION
**Pull request by momesgin**
### Summary
Moved the logic for generating a version based support link into its own helper, and reused it in the component and the unit tests.
Fixes rancher/dashboard#8657

### CODE REVIEW

File: shell/utils/__tests__/version.test.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    const version2 = 'v2.7.5-rc4';
+    const result2 = generateSupportLink(version2);
+
+    expect(result2).toStrictEqual(expectation);
+
+    const version3 = undefined;
+    const result3 = generateSupportLink(version3);
+
+    expect(result3).toStrictEqual(expectation);
+  });
```
Comment (rak-phillip) [inline shell/utils/__tests__/version.test.ts:29]: I think this test would be a perfect opportunity for `test.each` [^1]. So we should be able to write something along the lines of

```suggestion
const testCases = [
  ['v2.7-0bcf068e1237acafd4aca01385c7c6b432e22fd7-head', 'https://rancher.com/support-maintenance-terms'],
  ['v2.7.5-rc4', 'https://rancher.com/support-maintenance-terms'],
  [undefined, 'https://rancher.com/support-maintenance-terms'],
];

test.each(testCases)(
  'should generate support link corresponding to the version %s',
  (version, expected) => {
    const result = generateSupportLink(version);
    expect(result).toBe(expected);
  }
);
```

[^1]: https://jestjs.io/docs/api#testeachtablename-fn-timeout

**Review by rak-phillip — COMMENTED** [review]
I think this change looks good. I left a comment on a potential improvement that we can make to the test.

File: shell/utils/__tests__/version.test.ts (hunk 1 above)
Comment (momesgin) [reply shell/utils/__tests__/version.test.ts]: @rak-phillip That's a nice refactor! thanks!

**Review by rak-phillip — APPROVED** [review]
LGTM

**Review by rak-phillip — APPROVED** [review]

**Pull request merged**